### ISSUE TITLE
Exit Node Missing from Android App

### ISSUE THREAD

**Issue by W1BTR** [netbirdio/netbird#2056]
**Describe the problem**

According to #289 exit nodes have been added to the android app on Android. However, I dont see any options to select an exit node anywhere.

App version 0.0.22

**Comment by mlsmaycon** [comment]
Hello @W1BTR that is correct. Our android client does not support route selection yet. It will come in the next few weeks. 

For the time being, you can control exit-node distribution with access control policies and distribution groups.

**Comment by W1BTR** [comment]
Gotchya, thank you! Glad it's coming in the next few weeks. Apple tv is a must for me as well - Im guessing it's being slowly rolled out to all the apps.

Also, I love that by **default** exit nodes expose LAN devices too, since that's what I normally need.

**Comment by W1BTR** [comment]
Any updates on this? I see releases up to v 0.27, but android is still on 0.23.

**Comment by Mark3xtrm** [comment]
> Hello @W1BTR that is correct. Our android client does not support route selection yet. It will come in the next few weeks.
> 
> For the time being, you can control exit-node distribution with access control policies and distribution groups.

So, what's the state of this? Please, add exit nodes support for the Android app, it's something almost crucial and competitors like Tailscale/Heascale has it

**Comment by pscriptos** [comment]
> Exit node (client) has been implemented in Android for a long time now, there's just no selection in the client. You can assign an exit node in the mangement dashboard.

Is this also documented somewhere? It already works for my desktop app, but it doesn't work for my phone. Thank you.

**Comment by W1BTR** [comment]
Agreed. We shouldnt need to have people call in to an admin each time they need to switch to another network.

**Comment by nazarewk** [comment]
FYI: we are now focusing on rewriting Android client from React into a native Android application (it's really hard to give any kind of ETA) . We will be addressing any remaining Android issues and inconsistencies between versions after the rewrite.

**Comment by Mark3xtrm** [comment]
> FYI: we are now focusing on rewriting Android client from React into a native Android application (it's really hard to give any kind of ETA) . We will be addressing any remaining Android issues and inconsistencies between versions after the rewrite.

Thank you for the update! This is good news, especially as you are introducing these features. Good work and update us as soon as you can 🦾

**Comment by Anaxagoras-bc** [comment]
@nazarewk 

Is there any way to track the progress of the android app?  I just started setting up and testing netbird, but the lack of this feature has me staying with tailscale for now.

**Comment by nazarewk** [comment]
@DarkPhyber-hg I'm pretty sure https://github.com/netbirdio/android-client/tree/java is the branch.

We have just released the first version of the rewritten client as open test on Google Play, you can sign up for this at https://play.google.com/apps/testing/io.netbird.client

**Comment by Anaxagoras-bc** [comment]
Thank you!  If nothing else the new version actually connects without locking up on my phone!  I had to downgrade to an older version to even get my android phone to work with netbird.  Can't wait until they add support for selecting exit nodes.

**Comment by skyline-ch** [comment]
Hello, I'm testing Netbird and find it pleasant, even if I had trouble getting it to work properly at first, on all Android versions.
I installed the Beta 0.1.2 version. I can see the Networks tab, but there's only one resource.
I don't see the exit node, only the local network share.

I currently have the following infrastructure for testing:
- At home, a VM with Ubuntu is used to share the local 192.168.1.0/24 and the internet connection as an exit node.
- In a data center, I have a Shadow PC with Windows 11. I installed Netbird. I can access the local 192.168.1.0/24 network and use my home IP address as an internet connection.
- An Android smartphone that I connect via 5G for testing. It can connect to the local 192.168.1.0/24 network, but it can't use the exit node; I can't see it.

In the admin dashboard, the Win 11 PC and Android PC are in the same group with the same simple configuration.
Is there anything I can provide to help debug the problem, or perhaps I misunderstood a configuration that needs to be done?

The smartphone is a OnePlus Nord (AC2003) with Android 12

**Comment by b3nab** [comment]
> FYI: we are now focusing on rewriting Android client from React into a native Android application (it's really hard to give any kind of ETA) . We will be addressing any remaining Android issues and inconsistencies between versions after the rewrite.

I'm wondering why has been decided this way to have two mobile clients written in two different languages when there are battle tested frameworks like react-native which seems to solve the UI aspect quite well while keeping the possibility to integrate native and platform-specific code. 

Also (as far I understood looking at the code) the "core library" of the mobile apps is always a go module that is neither android nor ios default language.
There are maybe some other discussions/blog posts/articles/something already written that explains the choices behind the project?

**Comment by W1BTR** [comment]
> > FYI: we are now focusing on rewriting Android client from React into a native Android application (it's really hard to give any kind of ETA) . We will be addressing any remaining Android issues and inconsistencies between versions after the rewrite.
> 
> I'm wondering why has been decided this way to have two mobile clients written in two different languages when there are battle tested frameworks like react-native which seems to solve the UI aspect quite well while keeping the possibility to integrate native and platform-specific code.
> 
> Also (as far I understood looking at the code) the "core library" of the mobile apps is always a go module that is neither android nor ios default language. There are maybe some other discussions/blog posts/articles/something already written that explains the choices behind the project?

IIRC React native is awesome, but rather inefficient, which is especially noticeable on batter powered devices, and especially an issue on apps that would be running non-stop, such as Netbird.

**Comment by lixmal** [comment]
This is implemented now
